File: androidcontent/src/main/java/com/soul/androidcontent/MainActivity.java
package com.soul.androidcontent;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.soul.lib.test.TestActivity;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.bt_subassembly).setOnClickListener(this);
        findViewById(R.id.bt_customview).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.bt_subassembly:
                startActivity("com.soul.androidcontent.subassembly");
                break;
            case R.id.bt_customview:
                startActivity("com.soul.androidcontent.customview");
                break;
        }
    }


    private void startActivity(String packageName) {
        final Intent intent = new Intent(MainActivity.this, TestActivity.class);
        intent.putExtra("pageName", packageName);
        startActivity(intent);
    }
}

File: kotlin/src/main/java/com/soul/kotlin/MainActivity.kt
package com.soul.kotlin

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.soul.lib.test.TestActivity

class MainActivity : AppCompatActivity(), View.OnClickListener {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<TextView>(R.id.tv_kotlinBasics).setOnClickListener(this)
        findViewById<TextView>(R.id.tv_kotlinAdvance).setOnClickListener(this)
    }


    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.tv_kotlinBasics -> {
                startTextFragment("com.soul.kotlin.basics")
            }
            R.id.tv_kotlinAdvance -> {
                startTextFragment("com.soul.kotlin.advance")
            }
        }
    }

    fun startTextFragment(packageName: String) {
        val intent = Intent(this@MainActivity, TestActivity::class.java)
        intent.putExtra("pageName", packageName)
        startActivity(intent)
    }

}
